### PR TITLE
MINOR: Update Streams docs: quickstart and concepts

### PR DESCRIPTION
**Pull request by guozhangwang**
Added figures for topology and table-stream duality; added sections about aggregations; misc code fixes.

### CODE REVIEW

**Comment by sriramsub** [comment]
LGTM

File: docs/streams.html [hunk 1]
```diff
@@ -47,70 +50,66 @@
...
             <li>Designed as a <b>simple and lightweight client library</b>, which can be easily embedded in any Java application and integrated with any existing packaging, deployment and operational tools that users have for their streaming applications.</li>
             <li>Has <b>no external dependencies on systems other than Apache Kafka itself</b> as the internal messaging layer; notably, it uses Kafka's partitioning model to horizontally scale processing while maintaining strong ordering guarantees.</li>
-            <li>Supports <b>fault-tolerant local state</b>, which enables very fast and efficient stateful operations like joins and windowed aggregations.</li>
-            <li>Employs <b>one-record-at-a-time processing</b> to achieve low processing latency, and supports <b>event-time based windowing operations</b>.</li>
+            <li>Supports <b>fault-tolerant local state</b>, which enables very fast and efficient stateful operations like joins and aggregations.</li>
+            <li>Employs <b>one-record-at-a-time processing</b> to achieve millisecond processing latency, and supports <b>event-time based windowing operations</b> with late arrived data.</li>
```
Comment (dguy) [inline docs/streams.html]: late arriving data ??

File: docs/streams.html [hunk 2]
```diff
@@ -47,70 +50,66 @@
...
-            <li>A <b>stream processor</b> is a node in the processor topology; it represents a processing step to transform data in streams by receiving one input record at a time from its upstream processors in the topology, applying its operation to it, and may subsequently producing one or more output records to its downstream processors.</li>
+            <li>A <b>stream processing application</b> is any program that makes use of the Kafka Streams library. It defines its computational logic through one or more <b>processor topologies</b>, where a processor topology is a graph of stream processors (nodes) that are connected by streams (edges).</li>
+            <li>A <b>stream processor</b> is a node in the processor topology; it represents a processing step to transform data in streams by receiving one input record at a time from its upstream processors in the topology, applying its operation to it, and may subsequently producing one or more output records to its downstream processors. </li>
```
Comment (dguy) [inline docs/streams.html]: producing -> produce

File: docs/streams.html [hunk 3]
```diff
@@ -47,70 +50,66 @@
...
         </ul>
+        <img class="centered" src="/{{version}}/images/streams-concepts-topology.jpg">
+
 
         <p>
-        Kafka Streams offers two ways to define the stream processing topology: the <a href="#streams_dsl"><b>Kafka Streams DSL</b></a> provides
-        the most common data transformation operations such as <code>map</code> and <code>filter</code>; the lower-level <a href="#streams_processor"><b>Processor API</b></a> allows
-        developers define and connect custom processors as well as to interact with <a href="#streams_state">state stores</a>.
+            Kafka Streams offers two ways to define the stream processing topology: the <a href="#streams_dsl"><b>Kafka Streams DSL</b></a> provides
+            the most common data transformation operations such as <code>map</code> and <code>filter</code>, <code>join</code> and <code>aggregations</code> out of the box; the lower-level <a href="#streams_processor"><b>Processor API</b></a> allows
```
Comment (dguy) [inline docs/streams.html]: needs some extra commas, i.e, 
<code>map</code>, <code>filter</code>, <code>join</code>, and <code>aggregations</code>

**Review by dguy — COMMENTED** [review]
left a couple of minor comments relate to typos and grammar

File: docs/streams.html (hunk 1 above)
Comment (guozhangwang) [reply docs/streams.html]: ack. looking at other places I'd go with `late arrival of records` for consistency.

**Comment by guozhangwang** [comment]
Merged to trunk and 0.10.2.

**Pull request closed without merging**